packages/shared_preferences/lib/shared_preferences.dart

```diff
@@ -60,7 +64,17 @@ class SharedPreferences {
 
   /// Reads a set of string values from persistent storage,
   /// throwing an exception if it's not a string set.
-  List<String> getStringList(String key) => _preferenceCache[key] as List<String>;
+  List<String> getStringList(String key) =>
+      _preferenceCache[key] as List<String>;
+
+  /// Gets a JSON-encoded preference string and returns it as an Object. See
+  /// [JSON.decode] and [JsonDecoder] documentation for what happens when the
+  /// string is not a valid JSON string, and how to use the reviver.  Returns
+  /// null if the given key does not exist.
+  Object getObject(String key, {reviver(var key, var value)}) {
+    if (_preferenceCache[key] == null) return null;
```

Comment:
Ideally we should store the object without encoding it in _preferenceCache. Otherwise you are decoding it every time you access shared prefs even without an app restart. The only way to avoid that would be to have a secondary cache at app level which does not make much sense for sharedprefs.

I am also concerned that this API potentially would hide performance issues. SharedPreferences API was designed to be synchronous. This will cause user's app to miss frames if they try to store large objects.

Both of these issues can be fixed but would require a much larger change. How would you want to proceed?